pybamm/solvers/base_solver.py

```diff
@@ -1585,6 +1648,14 @@ def jacp(*args, **kwargs):
                 [t_casadi, y_and_S, p_casadi_stacked, v],
                 [jac_action_casadi],
             )
+            # Compute derivate wrt p-stacked (can be passed to solver to
+            # compute sensitivities online)
+            if return_jacp_stacked:
```

Comment:
should this go under the `elif model.calculate_sensitivities:` (line 1621)?

Reply:
Ah, yes. I've moved the code to a more appropriate place.